src/Runner.Common/RunServer.cs

```diff
@@ -67,10 +67,48 @@ private void CheckConnection()
             }
         }
 
-        public Task<AgentJobRequestMessage> GetJobMessageAsync(string id)
+        public Task<AgentJobRequestMessage> GetJobMessageAsync(string id, CancellationToken cancellationToken)
         {
             CheckConnection();
-            return _taskAgentClient.GetJobMessageAsync(id);
+            var jobMessage = RetryRequest<AgentJobRequestMessage>(async () =>
+                                                    {
+                                                        return await _taskAgentClient.GetJobMessageAsync(id, cancellationToken);
+                                                    }, cancellationToken);
+            return jobMessage;
+        }
+
+        private async Task<T> RetryRequest<T>(Func<Task<T>> func,
+                                                        CancellationToken cancellationToken,
+                                                        int maxRetryAttempts = 5
+                                                        )
+        {
+            var currentRetryAttempt = 1;
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                try
+                {
+                    return await func();
+                }
+                catch (Exception ex)
+                {
+                    Trace.Error("Catch exception during get full job message");
+                    Trace.Error(ex);
+
+                    if (currentRetryAttempt < maxRetryAttempts)
```

Comment:
NIT: I tried doing the math: with `currentRetryAttempt=1` and `maxRetryAttempts=5`, this code executes a maximum of 5 times. Technically, that's 1 try and 4 retries.

Reply:
Yeeeaaah 😅 , actually my goal was to run code 5 times, so it is 1 try + 4 retries. But I might miss the point and we should have 1 try + 5 retries. What do you think?